src/otx/v2/adapters/torch/mmengine/mmdet/modules/datasets/dataset.py

```diff
@@ -0,0 +1,154 @@
+                "ann_info": {"label_list": self.labels},
+                "ignored_labels": ignored_labels,
+                "bbox_fields": [],
+                "mask_fields": [],
+                "seg_fields": [],
+            }
+
+    def __init__(
+        self,
+        otx_dataset: DatasetEntity,
+        labels: list[LabelEntity],
+        pipeline: Sequence[dict],
+        test_mode: bool = False,
+        max_refetch: int = 1000,
+        **kwargs,
+    ) -> None:
+        """Initialize OTXDetDataset.
+
+        Args:
+            otx_dataset (DatasetEntiy): DatasetEntity from dataset api
+            labels (List[LabelEntity]): List of LabelEntity
+            pipeline (Sequence[dict]): List of data pipeline
+            test_mode (bool): Whether current dataset is for test or not
+            max_refetch (int): If ``Basedataset.prepare_data`` get a None img.
+                The maximum extra number of cycles to get a valid
+            kwargs: Additional kwargs
+        """
+        dataset_cfg = kwargs.copy()
+        _ = dataset_cfg.pop("org_type", None)
+        new_classes = dataset_cfg.pop("new_classes", [])
+        self.otx_dataset = otx_dataset
+        self.labels = labels
+        self.CLASSES = [label.name for label in labels]
+        self.domain = self.labels[0].domain
+        self.test_mode = test_mode
+        self.max_refetch = max_refetch
+
+        self._metainfo = {"classes": self.CLASSES, "domain": self.domain}
+
+        # Instead of using list data_infos as in BaseDetDataset, this implementation of dataset
+        # uses a proxy class with overriden __len__ and __getitem__; this proxy class
+        # forwards data access operations to otx_dataset.
+        # Note that list `data_infos` cannot be used here, since OTX dataset class does not have interface to
+        # get only annotation of a data item, so we would load the whole data item (including image)
+        # even if we need only checking aspect ratio of the image; due to it
+        # this implementation of dataset does not uses such tricks as skipping images with wrong aspect ratios or
+        # small image size, since otherwise reading the whole dataset during initialization will be required.
+        self.data_list: Any = OTXDetDataset._DataInfoProxy(otx_dataset, labels)
+
+        self.proposals = None  # Attribute expected by mmdet but not used for OTX datasets
+
+        if not test_mode:
+            self.img_indices = get_old_new_img_indices(self.labels, new_classes, self.otx_dataset)
+
+        self.pipeline = Compose(pipeline)
+        self.serialize_data = None  # OTX has own data caching mechanism
+        self._fully_initialized = False
+        self.full_init()
+
+    def full_init(self) -> None:
+        """OTXDetDataset do not have difference between full init and lazy init.
+
+        It is for compatibility with MMEngine
+        """
+        if self._fully_initialized:
+            return
+
+        self._fully_initialized = True
```

Comment:
Why is this needed? Do we do something like lazy init?

Reply:
MMengine BaseDataset does lazy init, I think I can remove this function.

Reply:
Change my implementation in daec976